core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -1490,11 +1490,21 @@
           (cond->> obc (wrap-with-order-by (r/$ obc -1)))
           (->> (wrap-with-top))))))
 
+(defn- app-time-extents->for-app-time [app-time-extents app-from-expr app-to-expr]
```

Comment:
I'd be keen to understand why we still need this case in the DML side - obv be good to get rid of it entirely from the SQL planner!